types/aws-lambda/trigger/lex-v2.d.ts

```diff
@@ -74,38 +68,41 @@ export interface LexV2SentimentScore {
 
 export interface LexV2SessionState {
     activeContexts?: LexV2ActiveContext[];
-    sessionAttributes?: { [key: string]: string };
+    sessionAttributes?: Record<string, string>;
     dialogAction?: LexV2DialogAction;
     intent: LexV2Intent;
     originatingRequestId: string;
 }
 
 export interface LexV2ActiveContext {
     name: string;
-    contextAttributes: { [key: string]: string };
+    contextAttributes: Record<string, string>;
     timeToLive: {
         timeToLiveInSeconds: number;
         turnsToLive: number
     };
 }
 
-export interface LexV2DialogAction {
-    slotToElicit?: string;
-    type: 'Close' | 'ConfirmIntent' | 'Delegate' | 'ElicitIntent' | 'ElicitSlot';
-}
+export type LexV2DialogActionType = 'Close' | 'ConfirmIntent' | 'Delegate' | 'ElicitIntent' | 'ElicitSlot';
 
-export interface LexV2ResultDialogAction extends LexV2DialogAction {
-    slotElicitationStyle?: 'Default' | 'SpellByLetter' | 'SpellByWord';
-}
+export type LexV2DialogAction =
+    | { type: Exclude<LexV2DialogActionType, 'ElicitSlot'> }
+    | { type: 'ElicitSlot', slotToElicit: string }
+    ;
+
+export type LexV2ResultDialogAction =
+    | Exclude<LexV2DialogAction, { type: 'ElicitSlot' }>
+    | (Extract<LexV2DialogAction, { type: 'ElicitSlot' }> & { slotElicitationStyle: 'Default' | 'SpellByLetter' | 'SpellByWord' })
```

Comment:
A little too fancy for me? Kind of a style nit, so take or leave it, but I'd prefer the alternatives explicitly defined in more boring interface style. That would mean defining the `LexV2DialogAction` alternatives explicitly too.

Reply:
I can switch to the following:

```
export type LexV2DialogAction =
    | { type: 'Close' }
    | { type: 'ConfirmIntent' }
    | { type: 'Delegate' }
    | { type: 'ElicitIntent' }
    | { type: 'ElicitSlot', slotToElicit: string }
    ;

export type LexV2ResultDialogAction =
    | { type: 'Close' }
    | { type: 'ConfirmIntent' }
    | { type: 'Delegate' }
    | { type: 'ElicitIntent' }
    | { type: 'ElicitSlot', slotToElicit: string, slotElicitationStyle: 'Default' | 'SpellByLetter' | 'SpellByWord' }
    ;
```
Which is more verbose but perhaps cleaner. Would you suggest adding `slotToElicit?: never` to each interface as suggested in the comment above? Alternatively, if we leave it the way I have it now, then it only needs to be written once.

Reply:
I was thinking more asking the lines of a union of `...WithSlot` / `...WithoutSlot` interfaces myself, but really it's just whatever's clearer and less likely to cause problems for users and future changes here.

Don't get too tied up here, if it makes sense to you it's fine.

Reply:
Added